flowmachine/flowmachine/core/query.py

```diff
@@ -862,6 +885,15 @@ def invalidate_db_cache(self, name=None, schema=None, cascade=True, drop=True):
             logger.debug("Dropping {}".format(full_name))
             with con.begin():
                 con.execute("DROP TABLE IF EXISTS {}".format(full_name))
+            q_state_machine.finish_resetting()
+        elif q_state_machine.is_resetting:
+            logger.debug(
+                f"Query '{self.md5}' is being reset from elsewhere, waiting for reset to finish."
+            )
+            while q_state_machine.is_resetting:
```

Comment:
Hmm, I'm not fully convinced we want to block here. I can see it being useful _if_ it is guaranteed that once we return the table has actually been removed, but I don't think we're checking that this has successfully happened So why not return immediately since we know the table is being dropped somewhere else?